pkg/cmd/roachtest/mixed_version_join_init.go

```diff
@@ -166,9 +165,9 @@ func unsuccessfullyAddNodeStep(nodes nodeListOption, joinNode int, newVersion st
 				// Today it includes both versions, which seems silly.
 				startArgs(fmt.Sprintf("-a=--join=%s", joinAddr)),
 			)
-			if !errors.Is(err, server.ErrIncompatibleBinaryVersion) {
-				t.Fatalf("expected err: %s, got %v", server.ErrIncompatibleBinaryVersion, err)
-			}
+			// TODO(adityamaru): Check whether we want to backport this particular
```

Comment:
@irfansharif this test wasn't backported to 20.1. The only incompatibility seems to be `ErrIncompatibleBinaryVersion` not being an error mode in 20.1. Just wanted to run this by you, whether this is fine as is?

Reply:
Skipping this test in 20.1 makes sense. It only makes sense for v20.2. It looks like the min version would bar this from running already. Is that not what you saw happen?

```
	r.Add(testSpec{
		Name:       "join-init/mixed",
		Owner:      OwnerKV,
		MinVersion: "v20.2.0",
```

Reply:
ahh sorry I didnt catch that. I'm just going to delete it in that case if that's fine?

Reply:
yea, that's more than fine.